code/datums/components/radioactive.dm

```diff
@@ -150,3 +152,6 @@ TYPEINFO(/datum/component/radioactive)
 
 		lines += "[ismob(owner) ? capitalize(he_or_she(owner)) : "It"] is [rad_word] with a [pick("fuzzy","sickening","nauseating","worrying")] [neutron ? "blue" : "green"] light.[examiner.job == "Clown" ? " You should touch [ismob(owner) ? him_or_her(owner) : "it"]!" : ""]"
 
+	/// Returns level of radioactivity (0 to 100) - note that SEND_SIGNAL returns 0 if the signal is not registered
+	proc/get_radioactivity()
+		return src.radStrength
```

Comment:
Kinda bad practice to return an (decimal) integer value here, since SEND_SIGNAL bitwise-ORs all the returns together. Doesn't necessarily matter right now, but it's something to keep in mind and is easier to clean up sooner rather than later.

Reply:
It should be okay for this, since radioactive component has `COMPONENT_DUPE_UNIQUE`

Reply:
But something else could register the signal.

Reply:
The proper, if cumbersome, way to do this is to pass something to hold the value (for example an empty list) as an argument to the signal and have the signal put the "return value" there. With a signal like this presumably you'd want this to be additive so it could check `return_list[1]` and if it exists it could `+=` to it.

Reply:
Alternatively, this particular thing could probably be done much better via atom properties, right? Instead of a signal the component just adds / removes / updates the atom propety on the thing it is attached to.

Reply:
I a-little-bit hate the idea of throwing an atom prop on stuff that has no behaviour by itself, especially since `radStrength` in this case is literally just the strength of this radioactive component.
Doing it as a list seems the least objectionable, but it's still a bit objectionable.